pkg/operators/v1/condition_types.go

```diff
@@ -0,0 +1,39 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OperatorConditionSpec allows a cluster admin to convey information about the state of an operator to OLM, potentially overriding state reported by the operator.
+type OperatorConditionSpec struct {
+	Overrides []Condition `json:"overrides,omitempty"`
```

Comment:
Should we add a TODO on the `Condition` type to migrate to `metav1.Condition` when we move everything to k8s 1.19+?

Is there anything that would prevent us from doing that? (e.g. a breaking change)

Reply:
We could use the [metav1.Condition](https://github.com/kubernetes/apimachinery/blob/release-1.19/pkg/apis/meta/v1/types.go#L1367-L1413) but we should be recognize that [OLM's condition type](https://github.com/operator-framework/api/blob/master/pkg/operators/v1/operator_types.go#L21-L39) has different fields:
* OLM's Condition has the `LastUpdateTime` field.
* The metav1.Condition has the `ObservedGeneration` field.

If we plan to update this struct to use the `metav1.Condition` we will have to handle any CRs that have set the `LastUpdateTime` field. Thoughts?

Reply:
I would definitely opt for using `metav1.Condition` directly. We should diverge as little as possible. If we find that `LastUpdateTime` is actually useful, then we can create a composite struct that embeds `metav1.Condition`.

Reply:
to add to this, I was planning on replacing the `Condition` struct you're using w/ `metav1.Condition` at some point down the road as well.